lang/c++/api/Node.hh

```diff
@@ -40,30 +40,38 @@ class GenericDatum;
 using NodePtr = std::shared_ptr<Node>;
 
 class AVRO_DECL Name {
+    struct Aliases;
+
     std::string ns_;
     std::string simpleName_;
+    std::unique_ptr<Aliases> aliases_;
```

Comment:
Originally, I simply had

```c++
std::vector<std::string> aliases_;
std::set<std::string> fqAliases_;
```

here but there is something peculiar going on when built under GCC 11 (as used in the GitHub action runner). Any addition of a container member here, with no other code change, causes some lifetime issues with the `weak_ptr` reference in `NodeSymbolic` that breaks certain `avrogencpp` tests. I have to suspect some issue with `libstdc++` 11.x because it's fine when built with Clang or if I take this new approach instead.

Reply:
@martin-g - if you could approve the workflows again, the tests should pass this time

Reply:
Done!